examples/reflection/type_data.rs

```diff
@@ -0,0 +1,154 @@
+//! The example demonstrates what type data is, how to create it, and how to use it.
+
+use bevy::prelude::*;
+use bevy::reflect::{FromType, TypeRegistry};
+
+fn main() {
```

Comment:
It's unusual to define traits, structs, and impls inside of `main` but I understand you did this to make it easy to read this example top to bottom. Maybe add a comment on how to read this here to comfort people who might be confused by this style.